src/components/Box.js

```diff
@@ -1,6 +1,22 @@
 import React, { Component } from 'react';
 
 export default class Box extends Component {
+
+    checkForLinks = item => {
+        let match = item.match(/((.*)\(([^\]]*)\)\[([^\]]*)\](.*))/);
+        if (match){
+            let link =
+                <a href={match[4]}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    key={item.length}>
+                    {match[3]}
+                </a>
+            item = [this.checkForLinks(match[2]), link, match[5]];
```

Comment:
Things are making more sense (I should have walked through the code more closely). You should be able to use the replace function with match references to make the whole algorithm more elegant https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter

Reply:
That's what I had initially though too, but with replace (and React) it will just treat it as a string. It would work with replace if dangerouslySetInnerHtml is used along with it, like this:
```
replaceLinks = item => {
    return item.replace(/\(([^\)]+?)\)\[([^\]]+?)\]/g, "<a href='$2'>$1</a>");
};

<p key={i} dangerouslySetInnerHTML={{ __html: this.replaceLinks(item) }} />
```

Reply:
Hmm. Good point. I'll leave it to you to make the call. Thanks for considering the notes and taking the time to find the best solution.

Reply:
I figure since this is totally populated by the dev and not a user this is probably a totally fine way to go about it.

Reply:
Had a very functional recursive regex designed to match parentheses perfectly... Only to realize that JavaScript's regex engine doesn't allow recursive patterns.. I'm wondering if some sort of custom markup syntax might be good for this and other purposes, using extremely unique symbols or something. Otherwise, links would probably remain a messy task.

Reply:
I think a basic regex is ok if it's reasonably robust. A lazy version of what you had using either Replace or the way you were already composing the string is fine for now at least. I just realized my example above was be interpolated which was probably confusing. It just shouldn't choke on something like `This (could be a [problem](https://www.example.com/) ) [ or [this]() ]`

Reply:
Does that make sense or am I not addressing something?

Reply:
Totally makes sense. I certainly began overthinking it and didn't try enough stuff. Enforcing standards such as making sure http|https|www|/ begin the url make it easier to avoid weird pitfalls, what do you think of this? [Regex101 attempt](https://regex101.com/r/XIyS5f/4)